includes/Core/Authentication/Authentication.php

```diff
@@ -126,6 +127,14 @@ final class Authentication {
 	 */
 	protected $first_admin;
 
+	/**
+	 * Has_Connected_Admins instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Has_Connected_Admins
+	 */
+	protected $has_connected_admins;
+
```

Comment:
One thing I just noticed is that as it is that the meta actions we're hooking on to in `Has_Connected_Admins` won't fire when a user disconnects because we use a direct DB query to delete all of their Site Kit meta all at once (I believe this is from when we used to call `disconnect` on many users at once, perhaps as part of the uninstall but it's out of scope to change that here).

Please add a call to `$this->has_connected_admins->delete()` in `Authentication::disconnect`.

Reply:
Updated.